pkg/discovery/dtofactory/group_dto_builder.go

```diff
@@ -63,77 +70,146 @@ func (builder *groupDTOBuilder) BuildGroupDTOs() []*proto.GroupDTO {
 			continue
 		}
 
-		// Groups for the Pod and containers belonging to the group
-		for etype, memberList := range entityGroup.Members {
+		// create groups for each configured entity type
+		for etype, resizeGroup := range GROUP_RESIZE_FLAG_MAP {
+			groupDTOs := builder.createGroupsByEntityType(entityGroup, etype, resizeGroup)
+
+			result = append(result, groupDTOs...)
+		}
+	}
+
+	return result
+}
+
+func (builder *groupDTOBuilder) createGroupsByEntityType(entityGroup *repository.EntityGroup,
+	entityType metrics.DiscoveredEntityType, resizeGroup bool) []*proto.GroupDTO {
 
-			// group id created using the parent type, name and target identifier
-			id := fmt.Sprintf("%s-%s[%s]", entityGroup.GroupId, builder.targetId, etype)
-			displayName := fmt.Sprintf("%ss By %s [%s]", etype, entityGroup.GroupId, builder.targetId)
+	var result []*proto.GroupDTO
 
-			var protoType proto.EntityDTO_EntityType
-			protoType, foundType := ENTITY_TYPE_MAP[etype]
-			if !foundType {
-				glog.Errorf("Invalid member entity type %s", etype)
-				continue
-			}
+	// member type
+	var protoType proto.EntityDTO_EntityType
+	protoType, foundType := ENTITY_TYPE_MAP[entityType]
+	if !foundType {
+		glog.Errorf("Invalid member entity type %s", entityType)
+		return []*proto.GroupDTO{}
```

Comment:
I believe there is no need to create and return an empty slice. Just return nil will work as Go allows you to append nil to a slice. There are several other places in the code that be treated the same way.